Kernel/InferenceStore.cpp

```diff
@@ -700,29 +699,37 @@ struct InferenceStore::TPTPProofPrinter
     if (rule==InferenceRule::INPUT) {
       vstring fileName;
       if (env.options->inputFile()=="") {
-	fileName="unknown";
+	      fileName="unknown";
       }
       else {
-	fileName="'"+env.options->inputFile()+"'";
+	      fileName="'"+env.options->inputFile()+"'";
       }
       vstring axiomName;
       if (!outputAxiomNames || !Parse::TPTP::findAxiomName(us, axiomName)) {
-	axiomName="unknown";
+	      axiomName="unknown";
       }
       inferenceStr="file("+fileName+","+quoteAxiomName(axiomName)+")";
     }
     else if (!parents.hasNext()) {
       vstring newSymbolInfo;
       if (hasNewSymbols(us)) {
-	newSymbolInfo = getNewSymbols("naming",us);
+        vstring newSymbOrigin;
+        if (rule == InferenceRule::FUNCTION_DEFINITION ||
```

Comment:
Would it make sense to introduce a checker function like `isFormulaTransformation`, `isSimplifyingInferenceRule`, etc? Although I'm not entirely convinced those cannot break if someone inserts a new value into `InferenceRule` into the wrong place... but introducing a new definition inference rule that is not added here might also break things, right? I'm actually not sure which one is better, what do you think?

Reply:
It would definitely make sense to do something more elegant here. But I couldn't think of anything easy, so I just decided to enhance the already quite bad rule of "if it has no parents it must be naming".

However, not much can ever get broken here. It's really just about what's written in the inference info of the TPTP version of the proof. We will have more opportunities for making this straight when users like Geoff bug us in the future about things like this one. At the moment, I don't think there is a clear understanding of what should be written here for each inference, and I don't think there is much software out there that would actually care.

Reply:
I agree, this particular condition probably stays like this until some major change in TPTP or SMTLIB, at which point we would do a more thorough sweep anyways.